dashboard/test/controllers/followers_controller_test.rb

```diff
@@ -183,55 +223,41 @@ class FollowersControllerTest < ActionController::TestCase
     assert_equal "Please enter a section code", flash[:alert]
   end
 
-  test "student can remove teacher" do
-    follower = @laurel_student_1
-
-    sign_in follower.student_user
+  test "remove has nice error when student does not actually have teacher" do
+    sign_in @laurel
 
-    assert_difference('Follower.count', -1) do
-      post :remove, student_user_id: follower.student_user.id, teacher_user_id: follower.user_id
+    assert_no_difference('Follower.count') do
+      post :remove, teacher_user_id: @chris.id
     end
-
-    assert !Follower.exists?(follower.id)
+    assert_redirected_to '/'
+    assert_equal "The teacher could not be found.", flash[:alert]
   end
 
-  test "student can remove teacher if teacher does not have email" do
+  test "student can remove teacher" do
     follower = @laurel_student_1
-    @laurel.update_attribute(:email, "")
 
     sign_in follower.student_user
 
     assert_difference('Follower.count', -1) do
-      post :remove, student_user_id: follower.student_user.id, teacher_user_id: follower.user_id
+      post :remove, teacher_user_id: follower.user_id
     end
 
     assert !Follower.exists?(follower.id)
   end
 
-  test "teacher can remove student" do
+  test "student can remove teacher if teacher does not have email" do
     follower = @laurel_student_1
+    @laurel.update_attribute(:email, "")
 
-    sign_in follower.user
+    sign_in follower.student_user
 
     assert_difference('Follower.count', -1) do
-      post :remove, student_user_id: follower.student_user_id, teacher_user_id: follower.user_id
+      post :remove, teacher_user_id: follower.user_id
     end
 
     assert !Follower.exists?(follower.id)
   end
 
-  test "student cannot remove other student" do
-    follower = @laurel_student_1
-
-    sign_in @student
-
-    assert_no_difference('Follower.count') do
-      post :remove, student_user_id: follower.student_user_id, teacher_user_id: follower.user_id
-    end
-    assert_response :forbidden
-    assert follower.reload # not deleted
-  end
```

Comment:
The student_user_id parameter is not used at all anymore (we look at current_user), it was left over from when remove was also used by teachers to remove students. Prob unlikely to accidentally regress.